### PR TITLE
pre-commit: test PR87143

### PR DESCRIPTION
**Pull request by dtcxzyw**
PR Link: https://github.com/llvm/llvm-project/pull/87143

### CODE REVIEW

File: bench/eastl/optimized/Int128_t.cpp.ll [hunk 1]
```diff
@@ -2428,20 +2428,14 @@ entry:
   %tempDivisor.sroa.8.0 = select i1 %cmp.i22, i64 %not.i2.i26, i64 %2
   %8 = or i64 %tempDivisor.sroa.8.0, %tempDivisor.sroa.0.0
   %9 = icmp eq i64 %8, 0
-  br i1 %9, label %if.then6, label %if.else
-
-if.then6:                                         ; preds = %entry
-  %10 = load i64, ptr %quotient, align 8
-  %div = udiv i64 %10, %tempDivisor.sroa.0.0
```
[the remark this answers is not in the document]
Comment (goldsteinn) [reply bench/eastl/optimized/Int128_t.cpp.ll:2435]: I believe it does. Is that why we now optimize this branch away?

[the remark this answers is not in the document]
Comment (dtcxzyw) [reply bench/eastl/optimized/Int128_t.cpp.ll:2435]: > Is that why we now optimize this branch away?

We can further mark `%if.then6` as unreachable.

[the remark this answers is not in the document]
Comment (goldsteinn) [reply bench/eastl/optimized/Int128_t.cpp.ll:2435]: Think so, although doesn't the essentially happen due to the immediate UB we detect in it?

[the remark this answers is not in the document]
Comment (dtcxzyw) [reply bench/eastl/optimized/Int128_t.cpp.ll:2435]: Yeah. But the current implementation just treats it as a deferred UB (`%div -> poison`).

**Pull request closed without merging**